### ISSUE TITLE
Short string - expected result not being returned 

### ISSUE THREAD

**Issue by tobyhede** [oramasearch/orama#328]
**Describe the bug**
I have a dataset of street types and abbreviations. See below for example

When I search for the abbreviation 'RD', the search only returns 'RDGE'

```
   {
    abbrv:  'RDGE',
    type:   'Ridge',
   },
   {
    abbrv:  'RD',
    type:   'Road',
   },
```

Search returns the wrong result, regardless of limit.
Search returns nothing when I use `exact`.

I have tried limiting the properties and using boost. 
I have also played with relevance but I have NFI what I am doing, so that didn't work. 

If I remove the entry for 'RDGE' then the search for 'RD' returns correctly. 

Interestingly, when I change the dataset and search does work, score for 'RD' is  higher than the score for 'RDGE' .

**Comment by tobyhede** [comment]
I have set up codesandbox demonstration:
https://codesandbox.io/s/lyrasearch-forked-3b8so4?file=/src/index.ts

**Comment by tobyhede** [comment]
Just noticed that if I change the order of the data in the initial load the system behaves as expected. 
ie have the "RD" abbreviation before 'RDGE'

**Comment by skoropadas** [comment]
@tobyhede looks like we have similar problems :) #327

**Comment by tobyhede** [comment]
I have done some more digging, I can replicate this behaviour using the smoke tests in the project

The following both works:
```
  
   await insert(db, { quote: 'AB' })
   await insert(db, { quote: 'ABCDEF' })

   const result = await search(db, { term: 'AB', exact: true})

    await insert(db, { quote: 'CDEF' })
    await insert(db, { quote: 'AB' })

   const result = await search(db, { term: 'AB', exact: true})
```

This does not work:
```
    await insert(db, { quote: 'ABCDEF' })
    await insert(db, { quote: 'AB' })

   const result = await search(db, { term: 'AB', exact: true})
```

**Comment by micheleriva** [comment]
Looking into this, thanks for the detailed issue and the repro

**Comment by micheleriva** [comment]
I can definitely reproduce the error, I'm on it!

**Comment by micheleriva** [comment]
Fixed in `v1.0.0-beta.7`, you can update from npm. Thanks again for opening this!
